### ISSUE TITLE
QUESTION: Does HMR for Server Actually Reload HTML Without Page Refresh

### ISSUE THREAD

**Issue by dtothefp** [jaredpalmer/razzle#1069]
I've been playing around with Razzle and how the Server HMR works. 

I generated a Razzle Project and run it
- `npx create-razzle-app my-app`
- `cd my-app && yarn start`

If I now edit the `src/Home.js` I see hot reloading working in the browser and I see the `server.js` being re-required by the `index.js`. But if I open my dev tools "view page source" my changes are not actually in the HTML. Is this expected behavior. If I reload it triggers the route compilation so I do see my HTML changes.

```
🔁  HMR Reloading `./server`...
[HMR] Updated modules:
[HMR]  - ./src/Home.js
[HMR]  - ./src/App.js
[HMR]  - ./src/server.js
[HMR] Update applied.
```

I've noticed in NextJS that the hot reloading of the server code actually will show up in the HTML without reloading the server.

**Comment by klarstrup** [comment]
As I recall NextJS "hot reloading" is actually just an autoreload, which would indeed reflect in the initial HTML show in "view source".

**Comment by nimaa77** [comment]
good question or bug report, I have the same problem with my after.js app too. 
I will get back to this issue soon.

**Comment by nimaa77** [comment]
https://github.com/jaredpalmer/razzle/issues/992#issuecomment-501119226

**Comment by zpetukhov** [comment]
#925 was linked to here and closed... not sure they are the same... so I commented there.
With my dirty hack to fix the development process.
https://github.com/jaredpalmer/razzle/issues/925#issuecomment-642313711

**Comment by fivethreeo** [comment]
Fixed in dev

**Comment by zpetukhov** [comment]
> Fixed in dev

@fivethreeo

Just upgraded to razzle 3.3.0, and I see the same behaviour.
HTML reloads in browser, but I see the old content.
Then I edit code again - html reloads - I see the first changes, do not see the latest changes.
In other words, the browser shows "one edit ago" app version.

Is it expected in 3.3.0?
If Yes - when it is expected to release? And how can I switch to dev version.

```
npx razzle -v
razzle, 3.3.0
```

**Comment by fivethreeo** [comment]
I should be more specific. Fixed in 4.0.0-finch :)

**Comment by fivethreeo** [comment]
https://github.com/jaredpalmer/razzle/issues/1377
